### ISSUE TITLE
Typed factories are rubbish

### ISSUE THREAD

**Issue by fir3pho3nixx** [castleproject/Windsor#410]
We need to fix:

 - https://github.com/castleproject/Windsor/issues/67 
 - https://github.com/castleproject/Windsor/issues/110
 - https://github.com/castleproject/Windsor/issues/130
 - https://github.com/castleproject/Windsor/issues/134
 - https://github.com/castleproject/Windsor/issues/325

This is blocking the memory leak on PR: https://github.com/castleproject/Windsor/pull/394

**Comment by jonorossi** [comment]
@fir3pho3nixx I'm thinking the first step would be to get failing unit tests for those issues into the repo and marked ignored so we can clearly see the current problems without having to dig through all those issues each time. Thoughts?

**Comment by fir3pho3nixx** [comment]
Yes, these were my sentiments exactly. Perhaps even setup a branch? I can do it if we agree.

**Comment by fir3pho3nixx** [comment]
Done. PR's should be raised against https://github.com/castleproject/Windsor/tree/typed-factories for this issue. 

I will add some of the failing tests I know about when I get time.

@mario-d-s feel free to jump in.

**Comment by mario-d-s** [comment]
Would love to, but on holiday till the 25th. I'll see where you stand with this by then, but I will follow progress in the meantime.


-------- Original Message --------

**Comment by mario-d-s** [comment]
@fir3pho3nixx Hey, I know it's been a while but I've had some busy times. I still intend to jump in here. What are the issues you're looking at and what would you recommend I try to tackle first?

**Comment by fir3pho3nixx** [comment]
Let’s scoop up some failing tests around this. Raise PR’s in the same way I did for PR #416. 

I have been debugging this in my spare time to gain more insight. I will be posting an opinionated/analysis piece at some point wrt. current state of play with this. From what I have seen so far I am leaning towards a rewrite where we double down on being more explicit and honour lifestyles as a first class citizen. This feature is very much needed, so we need to address this. Your help would be much appreciated.
